internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -163,6 +163,12 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 		return err
 	}
 
+	// fix for RHEL 6 hosts where ttyname returns a wrong terminal name
+	// (see process_linux.go for details)
+	if e.FileConfig.MountDev == "minimal" || e.EngineConfig.GetContain() {
```

Comment:
Will create a separate PR. This is a fix because actually the minimal check https://github.com/sylabs/singularity/blob/dbe15bdc62d81cbf6a45504aefefb3beddd25da0/internal/pkg/runtime/engine/singularity/process_linux.go#L71 has no effect as `singularity.conf` can't be parsed from container and the configuration is not passed between stage 1 and 2, so `e.EngineConfig.File.MountDev` is always equal to `yes` in stage 2